bindings/native/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+extern crate core;
+
+mod wrapper;
+mod shared;
+
+//#![allow(unused_imports)]
+use std::ffi::{CStr, CString};
+use std::cell::RefCell;
+use std::{ptr, slice};
+use std::error::Error;
+use std::os::raw::c_char;
+
+use crate::shared::hash_blake2b;
+use crate::wrapper::{StrongholdWrapper, WrapperError};
+
+thread_local!{
+    static LAST_ERROR: RefCell<Option<Box<dyn Error>>> = RefCell::new(None);
+}
+
+fn push_error(err: WrapperError)  {
+    LAST_ERROR.with(|prev| {
+        *prev.borrow_mut() = Some(Box::new(err));
+    });
+}
+
+pub extern "C" fn get_last_error() -> *const c_char {
+    let last_error = LAST_ERROR.with(|prev| prev.borrow_mut().take());
+
+    let last_error = match last_error {
+        Some(err) => err,
+        None => return ptr::null_mut(),
+    };
+
+    let s = CString::new(last_error.to_string()).unwrap();
+    s.as_ptr()
+}
+
+#[no_mangle]
+pub extern "C" fn create(snapshot_path_c: *const libc::c_char, key_c: *const libc::c_char) -> *mut StrongholdWrapper {
+    println!("[Rust] Create started");
+
+    let snapshot_path = unsafe { CStr::from_ptr(snapshot_path_c) };
+    let snapshot_path = snapshot_path.to_str().unwrap().to_string();
+    let key = unsafe { CStr::from_ptr(key_c) };
```

Comment:
I would be careful here, as the key pointer might also be allocated elsewhere and thus lying around in memory. I'd recommend to zeroize the key directly after passing it to Stronghold.